### PR TITLE
CEDS-3936 Display Cancellation Request Notification status on timeline

### PR DESCRIPTION
**Pull request by udinobi**

### CODE REVIEW

File: app/views/helpers/TimelineEvents.scala [hunk 1]
```diff
@@ -41,22 +42,23 @@ class TimelineEvents @Inject() (
...
+    }.sorted
 
-    val IndexToMatchForUploadFilesContent = sortedNotificationsSummaries.indexWhere(summary =>
-      summary.enhancedStatus == ADDITIONAL_DOCUMENTS_REQUIRED || summary.enhancedStatus == UNDERGOING_PHYSICAL_CHECK
-    )
-    val IndexToMatchForViewQueriesContent = sortedNotificationsSummaries.indexWhere(_.enhancedStatus == QUERY_NOTIFICATION_MESSAGE)
-    val IndexToMatchForFixResubmitContent = sortedNotificationsSummaries.indexWhere(_.enhancedStatus == ERRORS)
+    val notificationSummaries =
+      if (!allNotificationSummaries.exists(_.enhancedStatus == CANCELLED)) allNotificationSummaries
+      else allNotificationSummaries.filterNot(_.enhancedStatus == CUSTOMS_POSITION_GRANTED)
```
Comment (glidester) [inline app/views/helpers/TimelineEvents.scala:53]: I read this as: if we have received a CANCELLED notification return all the notifications (including the CUSTOMS_POSITION_GRANTED!) else return all notifications apart from CUSTOMS_POSITION_GRANTED.

I think we should always be filtering out CUSTOMS_POSITION_GRANTED as we never want to show this to the user in any circumstance.

**Review by glidester — CHANGES_REQUESTED** [review]

File: app/views/helpers/TimelineEvents.scala (hunk 1 above)
Comment (udinobi) [reply app/views/helpers/TimelineEvents.scala:53]: If we have **NOT** received a CANCELLED notification return all the notifications
else, If we have received a CANCELLED notification, return all notifications apart from CUSTOMS_POSITION_GRANTED

Comment (glidester) [reply app/views/helpers/TimelineEvents.scala:53]: Ah, missed the `!` at the start of the if! Yes but we still should not show the CUSTOMS_POSITION_GRANTED in any circumstances

Comment (udinobi) [reply app/views/helpers/TimelineEvents.scala]: I updated the ticket relatively to not show CUSTOMS_POSITION_GRANTED on the timeline.

File: app/models/declaration/submissions/EnhancedStatus.scala [hunk 2]
```diff
@@ -25,15 +25,21 @@ object EnhancedStatus extends Enumeration {
 
   implicit val format: Format[EnhancedStatus.Value] = EnumJson.format(EnhancedStatus)
 
+  implicit class InclusionInSet(status: EnhancedStatus) {
+    def in(set: Set[EnhancedStatus]): Boolean = set.contains(status)
+  }
```
[the remark this answers is not in the document]
Comment (udinobi) [reply app/models/declaration/submissions/EnhancedStatus.scala:30]: Nothing compelling. Still, the implicit method `in` gives you more flexibility when using placeholders in higher-order functions.

**Review by glidester — APPROVED** [review]

**Pull request merged**